src/rules/no-self-import.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileOverview Prohibits a file from importing itself
+ * @author Gio d'Amelio
+ */
+
+import resolve from 'eslint-module-utils/resolve'
+import isStaticRequire from '../core/staticRequire'
+
+function isImportingSelf(context, node, requireName) {
+  const filePath = context.getFilename()
```

Comment:
last I knew, Sublime uses `--stdin`, but with `--stdin-filename` (or something like that) to fake in the filename